### PR TITLE
Update deps, cleanup, and add router.

### PR DESCRIPTION
**Pull request by knowuh**
Getting the project ready for work on:

[#129960509]

https://www.pivotaltracker.com/n/projects/736901/stories/129960509

### CODE REVIEW

**Comment by knowuh** [comment]
Hello @dougmartin could you have a look at this?

These changes are to [allow teachers to reload the page without losing their place](https://www.pivotaltracker.com/story/show/129960509)

Some changes are simply updating deps, and the cascade from that (eg lowdash).

The rails-route changes are in 9280195

File: src/js/components/toc.coffee [hunk 1]
```diff
@@ -84,7 +85,7 @@ PageLink = React.createFactory React.createClass
   render: ->
```
Comment (dougmartin) [inline src/js/components/toc.coffee:82]: The local onClick handler code above can be removed.

File: src/js/components/nav_overlay.coffee [hunk 2]
```diff
@@ -12,6 +12,23 @@ Nav = React.createClass
   getInitialState: ->
     activity: null
 
+  componentDidUpdate: (prevProps) ->
+    if (prevProps.pageId != @props.pageId) || (prevProps.sequence != @props.sequence)
+      @openActivity()
+
+  openActivity: (pageId) ->
```
Comment (dougmartin) [inline src/js/components/nav_overlay.coffee]: The pageId parameter doesn't seem to be used in the method

File: src/js/components/app.coffee [hunk 3]
```diff
@@ -187,20 +186,26 @@ App = React.createClass
...
-    _.find pages, (p) => p.id == @state.pageId
+    _.find pages, (p) => p.id == @pageId()
 
   getQuestions: ->
     return [] unless @state.sequence
-    @getCurrentPage().questions || []
+    @getCurrentPage()?.questions || []
+
+  data: ->
+    Object.assign({}, @state, {pageId: @pageId()})
```
Comment (dougmartin) [inline src/js/components/app.coffee]: Should you use _.assign here?  Object.assign isn't supported in IE 11 - http://kangax.github.io/compat-table/es6/#test-Object_static_methods_Object.assign

**Comment by dougmartin** [comment]
Other than the Object.assign issue this looks good 👍

**Pull request merged**